app/models/miq_server/server_smart_proxy.rb

```diff
@@ -151,37 +147,22 @@ def forceVmScan
 
   # TODO: This should be moved - where?
   def is_vix_disk_supported?
-    caps = {:vixDisk => false}
+    # This is only available on Linux
+    return false unless Sys::Platform::IMPL == :linux
+
     begin
-      # This is only available on Linux
-      if Sys::Platform::IMPL == :linux
-        require 'VMwareWebService/VixDiskLib/VixDiskLib'
-        caps[:vixDisk] = true
-      end
+      require 'VMwareWebService/VixDiskLib/VixDiskLib'
+      return true
     rescue Exception
       # It is ok if we hit an error, it just means the library is not available to load.
     end
 
-    caps[:vixDisk]
+    false
   end
 
   def concurrent_job_max
-    update_capabilities
-    capabilities[:concurrent_miqproxies].to_i
-  end
-
-  def max_concurrent_miqproxies
     return 0 unless self.is_a_proxy?
-    MiqSmartProxyWorker.worker_settings[:count]
-  end
 
-  def update_capabilities
-    self.capabilities = {} if capabilities.nil?
-    # We can only update these values if we are working on the local server
-    # since they are determined by local resources.
-    if MiqServer.my_server == self
-      capabilities[:vixDisk] = self.is_vix_disk_supported?
-      capabilities[:concurrent_miqproxies] = max_concurrent_miqproxies
-    end
+    MiqSmartProxyWorker.worker_settings[:count].to_i
```

Comment:
Should this instead check the MiqServer's config instead of the local config?  Minor distinction, but calling `MiqServer.find(x).concurrent_job_max` shouldn't use the local servers settings, it should use the settings from the found server.  I'm not sure how important that is, cause I'm not sure how this method is used.

Reply:
Yeah, I really thought I did this :confused: I'll make the change.